airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_spec.py

```diff
@@ -1016,40 +1016,123 @@ def test_nested_group(mocker, connector_spec, should_fail):
+                    "select_type": {
+                        "type": "object",
+                        "display_type": "radio",
+                        "oneOf": [
+                            {
+                                "type": "object",
+                                "properties": {
+                                    "option_title": {"type": "string", "title": "Title", "const": "first option"},
+                                    "something": {"type": "string"},
+                                },
+                            },
+                            {
+                                "type": "object",
+                                "properties": {
+                                    "option_title": {"type": "string", "title": "Title", "const": "second option"},
+                                    "some_field": {"type": "boolean"},
+                                },
+                            },
+                        ],
+                    },
+                },
+            }, 
+            False,
         ),
+        # SUCCESS: display_type is the name of the property
         (
-            {"type": "object", "properties": {"jwt": {"type": "object", "properties": {"a": {"type": "string", "always_show": True}}}}},
+            {
+                "type": "object", 
+                "properties": {
+                    "display_type": {
+                        "type": "string",
+                    },
+                },
+            }, 
             False,
         ),
+        # FAILURE: display_type is set to an invalid value
         (
             {
-                "type": "object",
-                "properties": {"jwt": {"type": "object", "required": ["a"], "properties": {"a": {"type": "string", "always_show": True}}}},
-            },
+                "type": "object", 
+                "properties": {
+                    "select_type": {
+                        "type": "object",
+                        "display_type": "invalid",
+                        "oneOf": [
+                            {
+                                "type": "object",
+                                "properties": {
+                                    "option_title": {"type": "string", "title": "Title", "const": "first option"},
+                                    "something": {"type": "string"},
+                                },
+                            },
+                            {
+                                "type": "object",
+                                "properties": {
+                                    "option_title": {"type": "string", "title": "Title", "const": "second option"},
+                                    "some_field": {"type": "boolean"},
+                                },
+                            },
+                        ],
+                    },
+                },
+            }, 
             True,
         ),
+        # FAILURE: display_type is set on a non-oneOf field
```

Comment:
nit: We could also test whether display_type set to a valid value (dropdown/radio) on a non-oneOf property fails as expected

Reply:
Good idea, I agree that it's better to use a valid value in this test - I think this was a bad copy-paste. Will update this